packages/@guardian/source-react-components-development-kitchen/components/divider/Divider.tsx

```diff
@@ -1,18 +1,19 @@
 import { css, SerializedStyles } from '@emotion/react';
-
 import { border, text } from '@guardian/src-foundations/palette';
 import { space } from '@guardian/src-foundations';
 import { textSans } from '@guardian/src-foundations/typography';
-
 import { partialStyles, fullStyles } from './styles';
+
+type Sizes = 'full' | 'partial' | 'fit';
```

Comment:
`full` means stretch to the sides in an article (it touches the edges)
`partial` means stretch left but limit the length
`fit` means... don't stretch to touch the sides...

This api has grown over time, which is nice in a way - we're growing! - but it also really _feels_ like it did

Is there a case for making a some breaking changes (while we're still at such an early stage of adoption) to build a more valid api covering all cases?

Reply:
yeah agreed! I'd be interested in refactoring this component, as alluded to in the comment below (https://github.com/guardian/source/pull/1006#issuecomment-933388759).

What do you think about me putting through a followup PR to do that (and your other changes), as it's a bit beyond the scope of this one

Reply:
Can we do it the other way around? Do the refactor first and _then_ merge this PR? Like this we avoid the state where we avoid the need for a pinky promise to do things later (which may or may not happen regardless of all our best intentions)